python/lsst/ts/externalscripts/maintel/parameter_march_comcam.py

```diff
@@ -0,0 +1,217 @@
+# This file is part of ts_externalscripts
+#
+# Developed for the LSST Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["ParameterMarchComCam"]
+
+import asyncio
+import json
+import types
+
+import yaml
+from lsst.ts import salobj
+from lsst.ts.observatory.control.maintel.comcam import ComCam, ComCamUsages
+from lsst.ts.observatory.control.maintel.mtcs import MTCS
+
+from ..base_parameter_march import BaseParameterMarch
+
+
+class ParameterMarchComCam(BaseParameterMarch):
+    """Perform a parameter march by taking images at different
+    degree of freedom position with ComCam.
+
+    Parameters
+    ----------
+    index : int
+        Index of Script SAL component.
+    """
+
+    def __init__(self, index, descr="Perform a parameter march with ComCam.") -> None:
+        super().__init__(index=index, descr=descr)
+
+        self.mtcs = None
+        self.comcam = None
+
+        self.dz = 1500  # microns, offset for out-of-focus images
+
+        self.instrument_name = "LSSTComCam"
+
+    @property
+    def tcs(self):
+        return self.mtcs
+
+    @property
+    def camera(self):
+        return self.comcam
+
+    def set_metadata(self, metadata: salobj.type_hints.BaseMsgType) -> None:
+        """Sets script metadata.
+
+        Parameters
+        ----------
+        metadata : `salobj.type_hints.BaseMsgType`
+            Script metadata topic.
+        """
+
+        metadata.duration = (
+            self.config.n_steps
+            * 3
+            * (
+                self.config.exp_time
+                + self.camera.read_out_time
+                + self.camera.shutter_time
+            )
+        )
+
+        metadata.instrument = self.get_instrument_name()
+        metadata.filter = self.get_instrument_filter()
+
+    async def take_images(
```

Comment:
you should be able to implement this method in the base class.

Reply:
See comment above. Why would I not separate it, since ComCam will be taking triplets while LSSTCam takes only one image? Doesn't it make it cleaner as is?